meta/sqlite/sql.go

```diff
@@ -0,0 +1,71 @@
+package sqlite
+
+import (
+	"fmt"
+	"strings"
+)
+
+type QueryOpt func(int) (string, any)
+
+func (q QueryOpt) K(i int) string {
+	k, _ := q(i)
+	return k
+}
+
+func (q QueryOpt) V() any {
+	_, v := q(0)
+	return v
+}
+
+func Option(k, v any) QueryOpt {
+	return func(n int) (string, any) {
+		return fmt.Sprintf(" %s = $%d", k, n+1), v
+	}
+}
+
+func Code(v any) QueryOpt {
+	return Option("Code", v)
+}
+
+func Network(v any) QueryOpt {
+	return Option("Network", v)
+}
+
+func Station(v any) QueryOpt {
+	return Option("Station", v)
+}
+
+func Location(v any) QueryOpt {
+	return Option("Location", v)
+}
+
+func Mark(v any) QueryOpt {
+	return Option("Mark", v)
+}
+
+func Make(v any) QueryOpt {
+	return Option("Make", v)
+}
+
+func Model(v any) QueryOpt {
+	return Option("Model", v)
+}
+
+func Serial(v any) QueryOpt {
+	return Option("Serial", v)
+}
+
+func Sample(v any) QueryOpt {
+	return Option("Sample", v)
+}
+
+func ParseBool(str string) (bool, bool) {
```

Comment:
suggestion: could this go into an internal package? `meta/mark` uses almost identical parsing when decoding.
I also think that the function name should either change to `IgsParseBool`, or the function name stays the same, and you incorporate `strconv.ParseBool` (or add "1" "T" etc. directly), to justify the name - essentially, provide a better `ParseBool` function than `strconv.ParseBool`, and use that everywhere.

Reply:
good point, I'll have a think.  I'd like to use a standard true/false but we have a couple of times where it's true/false/maybe type settings.

Reply:
Have gone for the extend the function as above, to make it the same as the strconv version (i.e. add 0,1,t,f etc).